satpy/readers/geos_area.py

```diff
@@ -0,0 +1,146 @@
+"""Geostationary Projection / Area computations.
+
+This module computes properties and area definitions for geostationary
+satellites. It is designed to be a common module that can be called by
+all geostationary satellite readers and uses commonly-included parameters
+such as the CFAC/LFAC values, satellite position, etc, to compute the
+correct area definition.
+"""
+
+import numpy as np
+from pyresample import geometry
+
+
+def get_xy_from_linecol(line, col, offsets, factors):
+    """Get the intermediate coordinates from line & col.
+
+    Intermediate coordinates are actually the instruments scanning angles.
+    """
+    loff, coff = offsets
+    lfac, cfac = factors
+    x__ = float(col - coff) / (float(cfac) / 2**16)
+    y__ = float(line - loff) / (float(lfac) / 2**16)
+
+    return x__, y__
+
+
+def make_ext(ll_x, ur_x, ll_y, ur_y, h):
+    """Create the area extent from computed ll and ur.
+
+    Inputs (all in metres):
+    - ll_x: The lower left x coordinate
+    - ur_x: The upper right x coordinate
+    - ll_y: The lower left y coordinate
+    - ur_y: The upper right y coordinate
+    - h: The satellite altitude above the Earth's surface
+    Returns:
+    - aex: An area extent for the scene
+
+    """
+    aex = (np.deg2rad(ll_x) * h, np.deg2rad(ll_y) * h,
+           np.deg2rad(ur_x) * h, np.deg2rad(ur_y) * h)
+
+    return aex
+
+
+def get_area_extent(pdict):
+    """Get the area extent seen by a geostationary satellite.
+
+    Inputs:
+    - pdict: A dictionary containing common parameters:
+            nlines: Number of lines in image
+            ncols: Number of columns in image
+            cfac: Column scaling factor
+            lfac: Line scaling factor
+            coff: Column offset factor
+            loff: Line offset factor
+            scandir: 'N2S' for standard (N->S), 'S2N' for inverse (S->N)
+    Returns:
+    - aex: An area extent for the scene
+
+    """
+    # count starts at 1
+    cols = 1 - 0.5
+
+    if (pdict['scandir'] == 'S2N'):
+        lines = 0.5 - 1
+        scanmult = -1
+    else:
+        lines = 1 - 0.5
+        scanmult = 1
+    # Lower left x, y scanning angles in degrees
+    ll_x, ll_y = get_xy_from_linecol(lines * scanmult,
+                                     cols,
+                                     (pdict['loff'], pdict['coff']),
+                                     (pdict['lfac'], pdict['cfac']))
+
+    cols += pdict['ncols']
+    lines += pdict['nlines']
+    # Upper right x, y scanning angles in degrees
+    ur_x, ur_y = get_xy_from_linecol(lines * scanmult,
+                                     cols,
+                                     (pdict['loff'], pdict['coff']),
+                                     (pdict['lfac'], pdict['cfac']))
+    if (pdict['scandir'] == 'S2N'):
+        ll_y *= -1
+        ur_y *= -1
+
+    # Convert degrees to radians and create area extent
+    aex = make_ext(ll_x, ur_x, ll_y, ur_y, pdict['h'])
```

Comment:
I think it would be nice to use keyword arguments here, `make_ext(ll_x=ll_x, ...)`, to prevent confusion. I stumbled upon this because the argument order differs from the order in the returned extent.

Reply:
Changed now.